src/tools/time_marks.hh

```diff
@@ -48,7 +48,31 @@ public:
      *  - type_bc_change (hex 0x04)
      *  @see TimeMarks
      */
-    typedef unsigned long int Type;
+	struct Type {
+		Type() : bitmap_(0x1), equation_index_(1) {}
+		Type(unsigned long int bitmap, unsigned char equation_index) : bitmap_(bitmap), equation_index_(equation_index) {}
+
+		Type operator&(const Type& other) const {
+			//ASSERT(equation_index_ == other.equation_index_)((unsigned int)equation_index_)((unsigned int)other.equation_index_).error();
+			return Type( (bitmap_ & other.bitmap_), equation_index_);
+		}
+
+		Type operator|(const Type& other) const {
```

Comment:
- if equation index of one operand is zero, set equation index of the result to the other one.
- if equation indices differs (and none is zero) ... error